File: two grams prob.py
def most_frequent_twogram(s):
    freq = {}
    for i in range(len(s) - 1):
        pair = s[i:i+2]
        freq[pair] = freq.get(pair, 0) + 1

                    # Print all two-grams and their counts
    print("🔎 Frequency of all two-grams:")
    for k in sorted(freq):
        print(f"{k}: {freq[k]}")

                     # Find the most frequent two-gram
    most_common = max(freq, key=freq.get)
    return most_common

# Input
n = int(input("Enter string length: "))
s = input("Enter the string: ").strip()

# Output
print("Most frequent two-gram: ", most_frequent_twogram(s)) 

File: two grams prob.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    string s;
    cout << "Enter string length: ";
    cin >> n;
    cout << "Enter the string: ";
    cin >> s;

    unordered_map<string, int> freq;
    string max_pair;
    int max_count = 0;

    for (int i = 0; i < n - 1; ++i) {
        string pair = s.substr(i, 2);
        freq[pair]++;
        if (freq[pair] > max_count) {
            max_count = freq[pair];
            max_pair = pair;
        }
    }

    // Print all two-grams and their frequencies
    cout << "\n🔎 Frequency of all two-grams:\n";
    vector<string> keys;
    for (auto& p : freq)
        keys.push_back(p.first);
    sort(keys.begin(), keys.end());

    for (const auto& k : keys)
        cout << k << ": " << freq[k] << "\n";

    // Most frequent two-gram
    cout << "\n🏆 Most Frequent Two-Gram: " << max_pair << endl;

    return 0;
}

